cmd/posix-oneshot/main.go

```diff
@@ -57,33 +50,23 @@ func main() {
 	// Gather the info needed for reading/writing checkpoints
 	v := getVerifierOrDie()
 	s := getSignerOrDie()
-	origin := s.Name()
 
-	if *initialise {
-		// Create the directory structure and write out an empty checkpoint
-		if err := os.MkdirAll(*storageDir, dirPerm); err != nil {
-			klog.Exitf("Failed to create log directory: %q", err)
-		}
-		// TODO(mhutchinson): This empty checkpoint initialization should live in Tessera
-		emptyCP := &fmtlog.Checkpoint{
-			Origin: origin,
-			Size:   0,
-			Hash:   rfc6962.DefaultHasher.EmptyRoot(),
-		}
-		n, err := note.Sign(&note.Note{Text: string(emptyCP.Marshal())}, s)
-		if err != nil {
-			klog.Exitf("Failed to sign empty checkpoint: %s", err)
-		}
-		if err := posix.WriteCheckpoint(*storageDir, n); err != nil {
-			klog.Exitf("Failed to write empty checkpoint: %s", err)
+	if len(*entries) == 0 {
+		if *initialise {
+			_, err := posix.New(ctx, *storageDir, *initialise, tessera.WithCheckpointSignerVerifier(s, v))
+			if err != nil {
+				klog.Exitf("Failed to initialise storage: %v", err)
+			}
```

Comment:
Could this be simplified by instantiating the storage once (using the `*initialise` so it does the right thing), but then exiting immediately afterwards if `len(*entries) == 0`, or am I missing something?

I notice that the instantiations differ in using/not using the `WithBatching`, is there a side-effect there?

Reply:
The `WithBatching` uses the length of the `filesToAdd`, which requires evaluating `*entries`. If this isn't provided then `readEntriesOrDie` dies. I could make it so that function instead returns an error, and then we'd have to set the batch size to 0, and then remember to exit after init.... urgh. It's fewer lines of code that other way, but I think harder to follow.

Reply:
Ahh, right, yeah, that makes sense.

Separately, is it worth breaking the init lines out into a func? (coming from making the main func easy to grok for the common path)

Reply:
Merging this as is. We can revisit this if you feel strongly about it.